src/main/java/org/dcsa/bkg/service/impl/BKGServiceImpl.java

```diff
@@ -762,85 +766,48 @@ public Mono<BookingResponseTO> cancelBookingByCarrierBookingReference(
       BookingCancellationRequestTO bookingCancellationRequestTO) {
     OffsetDateTime updatedDateTime = OffsetDateTime.now();
     return getActiveBooking(carrierBookingRequestReference)
-        .flatMap(checkCancelBookingStatus)
+        .map(Booking::checkCancelBookingStatus)
         .flatMap(
             booking ->
-                bookingRepository
-                    .updateDocumentStatusAndUpdatedDateTimeForCarrierBookingRequestReference(
-                        bookingCancellationRequestTO.getDocumentStatus(),
-                        carrierBookingRequestReference,
-                        updatedDateTime)
-                    .flatMap(verifyCancellation)
-                    .thenReturn(booking))
+                performBookingCancellation(
+                    bookingCancellationRequestTO.getDocumentStatus(), updatedDateTime, booking))
         .map(
             booking -> {
               // Note the update to the version is applied to the active version. No new version is
               // created as a result of the cancelBookingByCarrierBookingReference
               booking.setDocumentStatus(ShipmentEventTypeCode.CANC);
+              ShipmentEvent shipmentEvent =
+                  shipmentEventMapper.shipmentEventFromBooking(
+                      booking, bookingCancellationRequestTO.getReason());
+              createShipmentEvent(shipmentEvent);
```

Comment:
I'm not sure if this will be triggered as there isn't anything subscribed to it. It would have to be chained along with something.

Reply:
changed accordingly